config/opal_check_pmi.m4

```diff
@@ -155,8 +155,21 @@ AC_DEFUN([OPAL_CHECK_PMI],[
 
     # enable Portals4 PMI if using Portals4 provider else slurm PMI check
     AS_IF([ test ! -z "$with_portals4"],[
-        AS_IF([test ! -z "$with_pmi" -a "$with_pmi" != "yes"],
-                [ompi_check_pmi_dir="$with_pmi"])
+            AS_IF([test ! -z "$with_pmi" -a "$with_pmi" != "yes"],
+                  AS_IF([test ! -z "$enable_pmi2" -a "$enable_pmi2" = "yes"],
+                        [slurm_pmi_found=no
+                        ompi_check_pmi_dir="$with_pmi"
+                        [OPAL_CHECK_PMI_LIB([$with_pmi],
```

Comment:
I think the square brackets around OPAL_CHECK_PMI_LIB are not necessary since this is already quoted by the square brackets that you just added.  Could you remove these from Lines 162 and 168 please?  Sorry to be so picky, these m4 macros can be a bit fragile.